CI/Dockerfile

```diff
@@ -34,63 +55,139 @@ RUN apt-get -y update; \
     pip install cython;
 
 
-# Copy scripts to docker image
-RUN mkdir -p /root/etc/
-COPY CI/ /root/etc/CI
-#TODO move sh file contents into this Dockerfile
-ENV docker_env=/root/etc/CI/env.sh
-
-ENV build_dir=/root/build_dir
-ENV install_dir=/root/opt
-
-
 FROM base as external_deps
 
-#setting the COMPILER variable
-ARG COMPILER=gcc
-ENV COMPILER=${COMPILER}
```

Comment:
Don't we still need to set this ENV?  Or at least need to do so for at least one of the packages that we build?

Reply:
The scripts use to accept a ```COMPILER``` argument that was then used to set ```CC``` and CXX``` arguments using some logic. 

if COMPILER == clang then change cc to clang and cxx to clang++
if COMPILER == gcc then change cc to gcc and cxx to g++

I've allowed the user to set CC and CXX directly and then we don't need the COMPILER arg